modules/gdscript/editor/gdscript_highlighter.cpp

```diff
@@ -473,11 +475,25 @@ Dictionary GDScriptSyntaxHighlighter::_get_line_syntax_highlighting_impl(int p_l
 
 		if (is_a_symbol) {
 			if (in_function_name) {
-				in_function_args = true;
-			}
-
-			if (in_function_args && str[j] == ')') {
-				in_function_args = false;
+				if (str[j] == '(') {
+					in_function_args = 1;
+					in_function_arg_dicts = 0;
+				}
+			} else if (in_function_args > 0) {
+				switch (str[j]) {
+					case '(':
+						in_function_args += 1;
+						break;
+					case ')':
+						in_function_args -= 1;
+						break;
+					case '{':
+						in_function_arg_dicts += 1;
+						break;
+					case '}':
+						in_function_arg_dicts -= 1;
+						break;
+				}
 			}
```

Comment:
Sorry, I realized that there is a bug in the code I suggested.

![](https://github.com/godotengine/godot/assets/47700418/5951406c-d46e-464f-8174-a21a5b11503b)

Here is the correct option:

```c++
			if (in_function_args > 0) {
				switch (str[j]) {
					case '(':
						in_function_args += 1;
						break;
					case ')':
						in_function_args -= 1;
						break;
					case '{':
						in_function_arg_dicts += 1;
						break;
					case '}':
						in_function_arg_dicts -= 1;
						break;
				}
			} else if (in_function_name && str[j] == '(') {
				in_function_args = 1;
				in_function_arg_dicts = 0;
			}
```

Reply:
No worries! I’ll get that updated in a few hours.